src/AspNetCore.Lambda/HttpHandlers.fs

```diff
@@ -275,4 +290,25 @@ let htmlFile (relativeFilePath : string) =
                 ctx
                 |> (setHttpHeader "Content-Type" "text/html"
                 >>= setBodyAsString html)
-        }
\ No newline at end of file
+        }
+
+let private handlerWithRootedPath (path:string) (handler : HttpHandler) = 
+    fun (ctx : HttpHandlerContext) ->
+        async {
+            let route = getRoute ctx.HttpContext
+            try
+                ctx.HttpContext.Items.Item RouteKey <- ((route |> Option.defaultValue "") + path)
+                return! handler ctx
+            finally
+                match route with
+                | Some route -> ctx.HttpContext.Items.Item RouteKey <- route
+                | None       -> ctx.HttpContext.Items.Remove RouteKey |> ignore
+        }
+
+let subPath (path:string) (handler : HttpHandler) =
+    route path >>=
```

Comment:
I think `route` currently compares if the passed in `path` matches exactly the value from `getRalative`, which would be initially `ctx.HttpContext.Request.Path.ToString()` and therefore will not succeed past this point.